apps/dash-brain-connectivity/app.py

```diff
@@ -0,0 +1,510 @@
+        # We either draw a new shape or we draw only the figure
+        if current_region is None:
+            return fig, f"removed {current_region}"
+        fig = hf.add_region_shape(fig, current_region)
+        return fig, f"{current_region}"
+
+    elif ctx.triggered[0]["prop_id"] == "val_slider.value":
+        # The threshold has changed
+        # Redraw the figure
+        fig = hf.make_heatmap(conn_mat_df)
+        if current_region is not None:
+            # We also want to draw the current region shape on the heatmap
+            fig = hf.add_region_shape(fig, current_region)
+        return fig, f"{threshold} and {current_region}"
+
+    else:
+        raise Exception(f"You shouldn't have come here: {ctx.triggered[0]['prop_id']}")
+
+
+# Draw circos here
+@app.callback(
+    Output("circos_graph", "children"),
+    [Input("val_slider", "value"), Input("store_selected_region", "data")],
+    [
+        State("store_conn_mat", "data"),
+        State("store_region_names", "data"),
+        State("threshold-mode", "value"),
+    ],
+)
+def connectivity_circos(threshold, current_region, conn_mat, region_names, thr_mode):
+    conn_mat = np.array(conn_mat)
+    conn_mat_df = hf.thr_conn_mat(conn_mat, threshold, region_names, thr_mode)
+
+    layout, highlight, connection = hf.make_circos(
+        conn_mat_df, focus_region=current_region
+    )
+
+    svg_size = 500
+    inner = 200
+    outer = 220
+    circos = dashbio.Circos(
+        size=svg_size,
+        layout=layout,
+        selectEvent={"0": "hover"},
+        config={"innerRadius": inner, "outerRadius": outer},
+        tracks=[
+            {
+                "type": "HIGHLIGHT",
+                "data": highlight,
+                "config": {
+                    "tooltipContent": {"name": "name"},
+                    "color": {"name": "color"},
+                    "innerRadius": inner,
+                    "outerRadius": outer,
+                },
+            },
+            {
+                "type": "CHORDS",
+                "data": connection,
+                "config": {
+                    "opacity": 0.5,
+                    "tooltipContent": {"name": "name"},
+                    "color": {"name": "color"},
+                },
+            },
+        ],
+    )
+
+    return circos
+
+
+# Highlight the table region
+@app.callback(
+    Output("region_table", "style_data_conditional"),
+    Input("store_selected_region", "data"),
+    prevent_initial_call=True,
+)
+def higlight_row(region_name):
+    """
+    When a region is selected, highlight the corresponding table row.
+    """
+    if region_name is not None:
+        return [
+            {
+                "if": {"filter_query": "{Region Name} = '%s'" % region_name},
+                "backgroundColor": "#3D9970",
+                "color": "white",
+            }
+        ]
+    return None
+
+
+if __name__ == "__main__":
+    app.run_server(debug=True, dev_tools_props_check=False, port=8052)
```

Comment:
```suggestion
    app.run_server(debug=True, dev_tools_props_check=False)
```

Reply:
I also use this trick of changing the port because I have several apps open at the same time, but let's keep it simple for the gallery